### PR TITLE
Do not append " - blocks" to the bitset label

### PR DESCRIPTION
**Pull request by dalg24**
Fixup for #6198

Rational:  Unnecessary as there is only one view (in contrast to UnorderedMap that has multiple accessory views).  IMO this is something that should rather be dealt with at call site.

code/label | Before this PR | After this PR
--- | --- | ---
`Bitset<D> bs(N);` | "Bitset - blocks" | "Bitset"
`Bitset<D> bs(view_alloc("MyLbl"), N);` | "MyLbl - blocks" | "MyLbl"
`UnorderedMap<K,V,D> m;` | "UnorderedMap - bitset - blocks" | "UnorderedMap - bitset"
`UnorderedMap<K,V,D> m(view_alloc("MyLbl"));` | "MyLbl - bitset - blocks" | "MyLbl - bitset"

cc @uliegecsm @romintomasetti

### CODE REVIEW

**Comment by romintomasetti** [comment]
I can't approve, but you get my approval :+1:

File: containers/src/impl/Kokkos_UnorderedMap_impl.hpp
Comment (masterleinad) [inline containers/src/impl/Kokkos_UnorderedMap_impl.hpp:44]: AFAICT the label exists whenever this function is called.
```suggestion
//! Append to the label.
template <typename... P>
auto append_to_label(const ViewCtorProp<P...>& view_ctor_prop,
                        const std::string& label) {
  static_assert (vcp_t::has_label); {
  ViewCtorProp<P...> new_ctor_props(view_ctor_prop);
  static_cast<ViewCtorProp<void, std::string>&>(new_ctor_props)
        .value.append(label);
  return new_ctor_props;
}
```

Comment (dalg24) [reply containers/src/impl/Kokkos_UnorderedMap_impl.hpp:44]: Feel free to push to my branch (make sure you format it properly)

**Review by romintomasetti — APPROVED** [review]

File: containers/src/impl/Kokkos_UnorderedMap_impl.hpp
Comment (fnrizzi) [inline containers/src/impl/Kokkos_UnorderedMap_impl.hpp:41]: just a thought: seems like this function (or variation of it) is being moved around where needed, 
and right here this is only used for unorderedmap. However this function has clearly no ties to unoredered map but just pertain to the ViewCtorProp. What about moving it closer to where ViewCtorProp is defined? It could be that, maybe, using this appending fucntion is something that could potentially be used elsewhere. No idea yet where, but keeping it closer to ViewCtorProp and maybe renaming it makes more sense to me and we don't have to keep moving it around.

**Review by fnrizzi — APPROVED** [review]
left a comment but not blocking on it

**Review by masterleinad — APPROVED** [review]

File: containers/src/impl/Kokkos_UnorderedMap_impl.hpp
Comment (dalg24) [reply containers/src/impl/Kokkos_UnorderedMap_impl.hpp:41]: I generally agree with your suggestion but I would rather have this tackled in some other follow up PR.
Are they other such functions that would need to be relocated?

Comment (fnrizzi) [reply containers/src/impl/Kokkos_UnorderedMap_impl.hpp:41]: no sure about other functions, i can try to look into it

**Pull request merged**
